aws/table_aws_elasticsearch_domain.go

```diff
@@ -40,6 +40,11 @@ func tableAwsElasticsearchDomain(_ context.Context) *plugin.Table {
 				Description: "The name of the domain.",
 				Type:        proto.ColumnType_STRING,
 			},
+			{
+				Name:        "engine_type",
```

Comment:
Oh, it is only available as an input 🤦‍♂️

https://github.com/aws/aws-sdk-go-v2/blob/service/elasticsearchservice/v1.19.3/service/elasticsearchservice/api_op_ListDomainNames.go#L36

Reply:
No, I didn't dream, it is here https://github.com/aws/aws-sdk-go-v2/blob/service/elasticsearchservice/v1.19.3/service/elasticsearchservice/types/types.go#L480

Reply:
However, it is only returned by the List Hydrate function. Is that case supported?

Reply:
Just figured I forgot to include an additional change for it to work. On it.

Reply:
Now it is working properly:

```
> select engine_type, elasticsearch_version from aws_elasticsearch_domain
+---------------+-----------------------+
| engine_type   | elasticsearch_version |
+---------------+-----------------------+
| Elasticsearch | 7.4                   |
| Elasticsearch | 6.7                   |
+---------------+-----------------------+
```

Reply:
Pushed the fix.

Reply:
Just realized the `aws_opensearch_domain` duplicates the results from the `aws_elasticsearch_domain` table because `EngineType` is not passed as an input to the API.

With the same connection:

```
> select engine_version from aws_opensearch_domain
+-------------------+
| engine_version    |
+-------------------+
| Elasticsearch_6.7 |
| Elasticsearch_7.4 |
+-------------------+
```

Reply:
Thanks @pdecat for the detailed analysis and the update 👍.

Does this https://github.com/turbot/steampipe-plugin-aws/blob/0c5faefb9d9b157ec845e1ba478f36ec172dcb19/aws/table_aws_elasticsearch_domain.go#L250-L251 change affect the other table columns?

Reply:
Pushed the fix to properly filter aws_elasticsearch_domain and aws_opensearch_domain tables on EngineType.

Reply:
Just to confirm @pdecat, are you able to query other columns like before without running into issues?

Reply:
Right now I'm facing an issue with the AWS plugin when re-building it: https://steampipe.slack.com/archives/C044P668806/p1690393064386099?thread_ts=1690283218.256439&cid=C044P668806

Reply:
Found the issue, `endpoint` was already in `aws_opensearch_domain`, just not where I expected it...

Reply:
It is working fine now.

Reply:
```
# steampipe query
Welcome to Steampipe v0.20.9
For more information, type .help
> select engine_type, elasticsearch_version from aws_elasticsearch_domain
+---------------+-----------------------+
| engine_type   | elasticsearch_version |
+---------------+-----------------------+
| Elasticsearch | 7.4                   |
| Elasticsearch | 6.7                   |
+---------------+-----------------------+
> select engine_type, engine_version from aws_opensearch_domain
+-------------+----------------+
| engine_type | engine_version |
+-------------+----------------+
+-------------+----------------+
```

Reply:
And with another connection containing opensearch domains:

```
# steampipe query
Welcome to Steampipe v0.20.9
For more information, type .help
> select engine_type, elasticsearch_version from aws_elasticsearch_domain
+-------------+-----------------------+
| engine_type | elasticsearch_version |
+-------------+-----------------------+
+-------------+-----------------------+
> select engine_type, engine_version from aws_opensearch_domain
+-------------+----------------+
| engine_type | engine_version |
+-------------+----------------+
| OpenSearch  | OpenSearch_1.3 |
+-------------+----------------+

```

Reply:
> Does this https://github.com/turbot/steampipe-plugin-aws/blob/0c5faefb9d9b157ec845e1ba478f36ec172dcb19/aws/table_aws_elasticsearch_domain.go#L250-L251 change affect the other table columns?

Nope.

Reply:
https://github.com/turbot/steampipe-plugin-aws/pull/1858#issuecomment-1680458119